### PR TITLE
GDB-10937 synchronize ask endpoint

### PR DESCRIPTION
**Pull request by boyan-tonchev**
## What
Synchronize UI models with the backend (BE).

## Why
The backend models have been changed.

## How
The UI models have been synchronized with the backend (BE).

### CODE REVIEW

File: src/js/angular/ttyg/templates/chat-item-detail.html [hunk 1]
```diff
@@ -6,73 +6,75 @@
...
-                </button>
-            </div>
-            <div class="derived-answer-hint"
-                 ng-if="loadingExplainResponse ||showActions && explainResponseModel.expanded">
-                <div>{{ 'ttyg.chat_panel.btn.derive_answer.hint' | translate }}</div>
-                <button class="btn deliver-answer-btn" ng-click="onAskHowAnswerWasDerived()">
-                    {{ 'ttyg.chat_panel.btn.derive_answer.label' | translate }}
-                </button>
+    <div class="answers" ng-repeat="answer in chatItemDetail.answers">
+        <div class="assistant" ng-if="answer">
```
Comment (svilenvelikov) [inline src/js/angular/ttyg/templates/chat-item-detail.html:10]: why the `ng-if=answer` was needed here?

File: src/js/angular/ttyg/templates/chat-item-detail.html [hunk 2]
```diff
@@ -6,68 +6,71 @@
...
-                                        gdb-tooltip="{{'ttyg.chat_panel.btn.open_in_sparql_editor.tooltip' | translate}}">
-                                    <i class="icon-sparql"></i>
-                                </button>
-                            </div>
+            <div class="assistant-message">
+                <div class="answer" ng-bind-html="MarkdownService.renderMarkdown(answer.message)"></div>
+                <div class="actions" ng-class="{'hidden-actions': !showActions || !$last}">
+                    <button class="btn btn-link btn-sm regenerate-question-btn"
+                            ng-click="regenerateQuestion()"
+                            ng-disabled="disabled"
```
Comment (svilenvelikov) [inline src/js/angular/ttyg/templates/chat-item-detail.html:19]: remove it

File: src/js/angular/ttyg/templates/chat-item-detail.html (hunk 1 above)
Comment (boyan-tonchev) [reply src/js/angular/ttyg/templates/chat-item-detail.html:10]: No, I will remove it.

File: test-cypress/stubs/ttyg/ttyg-stubs.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import {Stubs} from "../stubs";
+import {CHAT_MESSAGE_ROLE} from "../../../src/js/angular/models/ttyg/chat-message";
```
Comment (svilenvelikov) [inline test-cypress/stubs/ttyg/ttyg-stubs.js:2]: No, don't do that. Define constants in the test package and don't depend on the src

Comment (boyan-tonchev) [reply test-cypress/stubs/ttyg/ttyg-stubs.js]: Removed

File: src/js/angular/ttyg/templates/chat-item-detail.html (hunk 1 above)
Comment (boyan-tonchev) [reply src/js/angular/ttyg/templates/chat-item-detail.html]: done

**Review by svilenvelikov — DISMISSED** [review]

**Pull request merged**